### PR TITLE
DEVDOCS-3762: add-discount-to-checkout

### PR DESCRIPTION
**Pull request by traci707**
# [DEVDOCS-3762](https://jira.bigcommerce.com/browse/DEVDOCS-3762)

## What changed?
Added new endpoint, v3/checkouts/{checkoutId}/discounts

### CODE REVIEW

File: reference/checkouts.v3.yml [hunk 1]
```diff
@@ -145,6 +145,69 @@ paths:
...
+                    items:
+                      type: object
+                      properties:
+                        discounted_amount:
+                          type: number
+                          example: 10
+                        name:
+                          type: string
+                          example: manual
+                        id:
```
Comment (valentindellangela) [inline reference/checkouts.v3.yml:189]: This api only accepts `discounted_amount` and `name` in the payload.

File: reference/checkouts.v3.yml [hunk 2]
```diff
@@ -145,6 +145,69 @@ paths:
...
+        in: path
+        required: true
+    post:
+      summary: Add Discount to Checkout
+      operationId: post-store_hash-v3-checkouts-checkoutId-discounts
+      responses:
+        '200':
+          $ref: '#/responses/CheckoutResponse'
+      description: |-
+        Adds a discount to an existing *checkout*.
```
Comment (valentindellangela) [inline reference/checkouts.v3.yml:165]: Maybe we can add that this discount only applies to `line_items`?

**Comment by traci707** [comment]
@valentindellangela Can you look at my changes?

File: reference/checkouts.v3.yml [hunk 3]
```diff
@@ -1747,33 +1810,10 @@ parameters:
     enum:
       - consignments.available_shipping_options
 securityDefinitions:
-  X-Auth-Token:
+  Checkouts:
+    name: API Key
```
Comment (slsriehl) [inline reference/checkouts.v3.yml:1814]: based on the endpoints, this looks like a store-management API. if that's true, the OAuth stuff needs to go back in, and the access_token gets passed as the X-Auth-Token.

```suggestion
    name: X-Auth-Token
```

**Review by slsriehl — CHANGES_REQUESTED** [review]
looks like the auth section may need to be cleaned up. depending on the rest of the file, this may be duplication that's correctly removed. please confirm and take appropriate action, if any.

@traci707

**Review by markcmurphy — APPROVED** [review]

**Pull request merged**